ArticleTemplates/assets/js/boot.js

```diff
@@ -158,6 +158,45 @@ const init = opts => {
                 common.init();
             });
         }
+
+        const listenForEmailEmbedIFrameResize = () => {
+            const allowedOrigins = ['https://www.theguardian.com'];
+
+            const allIframes = [].slice.call(
+                document.querySelectorAll('.email-sub__iframe')
+            );
+            window.addEventListener('message', (event) => {
+                if (!allowedOrigins.includes(event.origin)) return;
+
+                const iframes = allIframes.filter((i) => {
+                    try {
+                        return i.contentWindow === event.source;
+                    } catch (e) {
+                        return false;
+                    }
+                });
+                if (iframes.length !== 0) {
+                    try {
+                        const message = JSON.parse(event.data);
+                        switch (message.type) {
+                        case 'set-height': {
+                            const value = parseInt(message.value);
+                            if (!Number.isInteger(value)) return;
+
+                            iframes.forEach((iframe) => {
+                                iframe.height = `${value}`;
```

Comment:
Listening for `message` from an iframe can introduce security risks (https://developer.mozilla.org/en-US/docs/Web/API/Window/postMessage#security_concerns). The main two to consider are:

## Verifying the sender's identity

It is important to ensure you only respond to messages originating from known sources. We have achieved this by:

* Discarding the message if it does not come from a known origin (L169)
* Checking that the `iframe.contentWindow === event.source`

## Verifying the syntax of the message

If a message from an unknown source was processed, it could contain a payload that enables XSS. **Especially** in our case, as we are setting an attribute's value from the payload. We have mitigated this by:

* Parsing the message into a known format (so it has a `type` and `value`)
* using `parseInt` to extract the height value from the message, and discarding the message if the parsed value is not an integer
* only then setting the `height` attribute with the parsed value

no other code path will update the attribute's value

### Summary

There are valid security concerns around listening to messages from iframes, and we have taken appropriate mitigations against them.